File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
    const numSet = new Set(nums);
    console.log({nums, head})
    let node = head;
    let prev = new ListNode(0, head);
    let ret = prev;
    while (node) {
        if (numSet.has(node.val)) {
            prev.next = node.next;
            node.next = null;
            node = prev.next;
        } else {
            prev = node;
            node = node.next;
        }
    }
    return ret.next;
};

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        del_nums = set(nums)

        root = None
        node = head
        tail = root
        
        while node:
            value = node.val 
            if value in del_nums:
                node = node.next
                continue
            if not root:
                root = ListNode(value)
                tail = root
            else:   
                tail.next = ListNode(value)
                tail = tail.next
            node = node.next
        return root